design/one-pager-controller-options-overrides.md

```diff
@@ -0,0 +1,42 @@
+# Controller Options Overrides
+
+- Owners:
+  - @max-melentyev
+- Reviewers: Crossplane Maintainers
+- Status: draft
+
+## Background
+
+There are cases when single configuration doesn't work for all controllers of a provider.
+Different Crossplane controllers may have very different load profile and number of managed resources.
+Consider examples in crossplane-contrib/provider-aws:
+
+- Reconciling route53 records is fast, but having many resources of this type may trigger AWS API
+throttling when all of them are reconciled with the same interval as other resources.
+- Reconciling EC2 instance is relatively slow, and having many managed instances can
+build up a long queue.
+
+Having a feature that makes it possible to override default poll interval for a specific controller
+would allow to avoid these issues without affecting other responsiveness of controllers in the provider.
```

Comment:
I wonder if there's anything else we'd want to override, apart from poll interval.

I'm thinking things like configuring exponential backoff for retries, or number of worker goroutines for a particular controller.

Interestingly poll interval feels like it could be configured on a per-MR basis, whereas the others are more a per-MR-type / per controller basis.